FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -1648,31 +2279,64 @@ BaseType_t xReturn;
 	UDPPacket_t *pxUDPPacket;
 	IPHeader_t *pxIPHeader;
 	UDPHeader_t *pxUDPHeader;
+	NetworkEndPoint_t *pxEndPoint = prvFindEndPointOnNetMask( pxNetworkBuffer );
 	size_t uxDataLength;
 
 		pxUDPPacket = ipPOINTER_CAST( UDPPacket_t *, pxNetworkBuffer->pucEthernetBuffer );
 		pxIPHeader = &pxUDPPacket->xIPHeader;
-		pxUDPHeader = &pxUDPPacket->xUDPHeader;
-		/* HT: started using defines like 'ipSIZE_OF_xxx' */
-			pxIPHeader->usLength				= FreeRTOS_htons( ( uint16_t ) lNetLength + ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_UDP_HEADER );
-		/* HT:endian: should not be translated, copying from packet to packet */
-		pxIPHeader->ulDestinationIPAddress = pxIPHeader->ulSourceIPAddress;
-		pxIPHeader->ulSourceIPAddress	   = *ipLOCAL_IP_ADDRESS_POINTER;
-		pxIPHeader->ucTimeToLive		   = ipconfigUDP_TIME_TO_LIVE;
-		pxIPHeader->usIdentification	   = FreeRTOS_htons( usPacketIdentifier );
-		usPacketIdentifier++;
-		pxUDPHeader->usLength			   = FreeRTOS_htons( ( uint32_t ) lNetLength + ipSIZE_OF_UDP_HEADER );
-		vFlip_16( pxUDPHeader->usSourcePort, pxUDPHeader->usDestinationPort );
 
-		/* Important: tell NIC driver how many bytes must be sent */
-		uxDataLength = ( ( size_t ) lNetLength ) + ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_UDP_HEADER + ipSIZE_OF_ETH_HEADER;
+	#if( ipconfigUSE_IPv6 != 0 )
+		if( ( pxIPHeader->ucVersionHeaderLength & 0xf0U ) == 0x60U )
+		{
+		UDPPacket_IPv6_t *xUDPPacket_IPv6;
+		IPHeader_IPv6_t *pxIPHeader_IPv6;
+
+			xUDPPacket_IPv6 = ipPOINTER_CAST( UDPPacket_IPv6_t *,pxNetworkBuffer->pucEthernetBuffer );
+			pxIPHeader_IPv6 = &( xUDPPacket_IPv6->xIPHeader );
+			pxUDPHeader = &xUDPPacket_IPv6->xUDPHeader;
+
+			pxIPHeader_IPv6->usPayloadLength = FreeRTOS_htons( ( uint16_t ) lNetLength + ipSIZE_OF_UDP_HEADER );
+
+			{
+				( void ) memcpy( pxIPHeader_IPv6->xDestinationAddress.ucBytes, pxIPHeader_IPv6->xSourceAddress.ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+				( void ) memcpy( pxIPHeader_IPv6->xSourceAddress.ucBytes, pxEndPoint->ipv6_settings.xIPAddress.ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+			}
+
+			xUDPPacket_IPv6->xUDPHeader.usLength = FreeRTOS_htons( ( uint16_t ) lNetLength + ipSIZE_OF_UDP_HEADER );
+			vFlip_16( pxUDPHeader->usSourcePort, pxUDPHeader->usDestinationPort );
+			uxDataLength = ( size_t ) lNetLength + ipSIZE_OF_IPv6_HEADER + ipSIZE_OF_UDP_HEADER + ipSIZE_OF_ETH_HEADER;
+		}
+		else
+	#endif /* ipconfigUSE_IPv6 */
+		{
+			pxUDPHeader = &pxUDPPacket->xUDPHeader;
+			/* HT: started using defines like 'ipSIZE_OF_xxx' */
+			pxIPHeader->usLength				= FreeRTOS_htons( ( uint16_t ) lNetLength + ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_UDP_HEADER );
+			/* HT:endian: should not be translated, copying from packet to packet */
```

Comment:
Maybe this comment can be modified as well...

Reply:
"HT" comment removed. This one was very old.

Reply:
Thanks